src/Interactive/EditorFeatures/Core/Extensibility/Interactive/InteractiveEvaluator.cs

```diff
@@ -260,50 +260,58 @@ private Dispatcher Dispatcher
 
         private void SubmissionBufferAdded(object sender, SubmissionBufferAddedEventArgs args)
         {
-            AddSubmission(_currentWindow.TextView, args.NewBuffer);
+            AddSubmission(_currentWindow.TextView, args.NewBuffer, this.LanguageName);
         }
 
         // The REPL window might change content type to host command content type (when a host command is typed at the beginning of the buffer).
         private void LanguageBufferContentTypeChanged(object sender, ContentTypeChangedEventArgs e)
         {
             var buffer = e.Before.TextBuffer;
-            var afterIsCSharp = e.AfterContentType.IsOfType(ContentTypeNames.CSharpContentType);
-            var beforeIsCSharp = e.BeforeContentType.IsOfType(ContentTypeNames.CSharpContentType);
 
-            if (afterIsCSharp == beforeIsCSharp)
+            var afterIsLanguage = e.AfterContentType.IsOfType(this.ContentType.TypeName);
+            var afterIsInteractiveCommand = e.AfterContentType.IsOfType(PredefinedInteractiveCommandsContentTypes.InteractiveCommandContentTypeName);
+            var beforeIsLanguage = e.BeforeContentType.IsOfType(this.ContentType.TypeName);
+            var beforeIsInteractiveCommand = e.BeforeContentType.IsOfType(PredefinedInteractiveCommandsContentTypes.InteractiveCommandContentTypeName);
+
+            if (afterIsLanguage == beforeIsLanguage ||
```

Comment:
This would be potentially problematic if there's a third content type that we can transition to or from. Consider the content types of C#, Interactive, and Foo. If we transition from Foo -> Interactive, since afterIsLanguage == beforeIsLanguage == false, this won't fire.

I'm not sure if this scenario exists though.